flow-model-generator/modules/flow-model-generator-core/src/main/java/io/ballerina/flowmodelgenerator/core/AgentsGenerator.java

```diff
@@ -283,8 +283,9 @@ public JsonElement genTool(JsonElement node, String toolName, String connectionN
         String path = flowNode.metadata().icon();
         if (nodeKind == NodeKind.FUNCTION_DEFINITION) {
             sourceBuilder.token()
-                    .name("@ai:AgentTool").
-                    name(System.lineSeparator());
+                    .descriptionDoc(description)
```

Comment:
We have to set the description only if description is provided, please check the function `genDescription`.

Reply:
fixed